packages/context/src/resolver.ts

```diff
@@ -154,6 +154,7 @@ function resolve<T>(
           // If the `optional` flag is set for the injection, the resolution
           // will return `undefined` instead of throwing an error
           optional: injection.metadata.optional,
+          asProxyWithInterceptors: injection.metadata.asProxyWithInterceptors,
```

Comment:
Similar comment here: now that metadata are inheriting from resolution options, can we simply copy all metadata?

```ts
return ctx.getValueOrPromise(key, {
  ...(injection.metadata),
  session: s,
});
```

Please extract all parts related to this inheritance change into a standalone pull request, so that we can land them faster and more importantly to get a proper entry in the change-log.

Reply:
Please note we use `es2017` but the object spread syntax is new in `es2018`.

Reply:
I tried with Node 8 and 10 and both work. Updated.